runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -163,10 +165,35 @@ void registerTable(@Nonnull String namespace,
 
         TableMetadata.Builder metadataBuilder = TableMetadata.newBuilder();
         metadataBuilder.setDiskBased(tableOptions.getPersistentDataPath().isPresent());
+
+        // Schema validation to ensure that there is either proper modification of the schema across open calls.
+        // Or no modification to the protobuf files.
+        boolean hasSchemaChanged = false;
+        CorfuRecord<TableDescriptors, TableMetadata> oldRecord = this.registryTable.get(tableNameKey);
+        if (oldRecord != null) {
+            if (!oldRecord.getPayload().getFileDescriptorsMap().equals(tableDescriptors.getFileDescriptorsMap())) {
+                hasSchemaChanged = true;
+                int oldTableEntries = tableMap.get(
+                        TableRegistry.getFullyQualifiedTableName(namespace, tableName)
+                ).count();
+                log.error("registerTable: Schema update detected for table "+namespace+" "+ tableName);
```

Comment:
The this be log.warn?

Reply:
I kept it as error because it is unsupported at this time and folks should really never see this.
In next release, we can move this to info once a migration path has been established.
Does that sound ok?

Reply:
Sure.